### PR TITLE
Hotfix/JM-7788 Juror search name fix

### PR DESCRIPTION
**Pull request by akikrahman1**
### JIRA link (if applicable) ###

https://centralgovernmentcgi.atlassian.net/browse/JM-7788

### Change description ###

Fix juror name search to allow for an Initial in the name

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by CCHAPMAN01 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/juror/api/moj/repository/JurorRepositoryImpl.java [hunk 1]
```diff
@@ -83,8 +83,16 @@ public JPAQuery<Tuple> fetchFilteredJurorRecords(JurorRecordFilterRequestQuery q
...
 
         if (null != query.getJurorName()) {
-            partialQuery.where(JUROR.firstName.concat(" ").concat(JUROR.lastName).containsIgnoreCase(
-                query.getJurorName()));
+
+            String[] names = query.getJurorName().split(" ");
+
+            if (names.length == 2) {
+                partialQuery.where(JUROR.firstName.containsIgnoreCase(names[0])
+                    .and(JUROR.lastName.containsIgnoreCase(names[1])));
```
Comment (CCHAPMAN01) [inline src/main/java/uk/gov/hmcts/juror/api/moj/repository/JurorRepositoryImpl.java:91]: some last names have spaces, perhaps we need to see if `names.length >1 `then for each array item in the search criteria, then check whether either first name of last name contains any item in the list?

File: src/integration-test/resources/db/JurorRecordController_searchForJurorRecords.sql [hunk 2]
```diff
@@ -19,7 +20,8 @@ VALUES ('641600091', 'MR','LNAMEFIVEFOURZERO','FNAMEFIVEFOURZERO', '1989-03-31',
 ('641500098', 'MR','Lnamenineeight','Fnamenineeight', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true),
 ('641500099', 'MR','Lnameninenine','Fnameninenine', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true),
 ('641500100', 'MR','Lnamenineten','Fnamenineten', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true),
-('641500101', 'MR','Lnamenineten','Fnamenineten', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true);
+('641500101', 'MR','Lnamenineten','Fnamenineten', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true),
+('641700120', 'MR','Lnametwo','Fnametwo I', '1988-01-01', '543 STREET NAME', 'ANYTOWN', 'CH1 2AN', true);
```
Comment (CCHAPMAN01) [inline src/integration-test/resources/db/JurorRecordController_searchForJurorRecords.sql:24]: Will need another test case for a surname with a space in it

Comment (akikrahman1) [reply src/integration-test/resources/db/JurorRecordController_searchForJurorRecords.sql]: This gets a bit messy, the issue raised is the users dont want to have to enter the initial so will cover that in the updated code.

File: src/main/java/uk/gov/hmcts/juror/api/moj/repository/JurorRepositoryImpl.java (hunk 1 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/repository/JurorRepositoryImpl.java:91]: covering the case where we match the first part of the firstname, then the last name can be multipart

**Comment by akikrahman1** [comment]
The solution here will not be perfect as its not possible to detect where the firstname ends and lastname starts on the name field. have tried to cover the most used scenarios. the best solution is to split the firstname and lastname into two fields.

**Pull request merged**